README.md

```diff
@@ -1,37 +1,129 @@
+
+> Production ready DuckDB Node.js library written in TypeScript.
+> [<img src="https://www.deepcrawl.com/wp-content/themes/deepcrawl/images/deepcrawl-logo.svg" height="200" width="300" align="right">](https://www.deepcrawl.com/)
+
 ## Overview
 
 - This is a library that adds support for [DuckDB](https://duckdb.org/) to NodeJS.
-- It comes preinstalled with DuckDB ver 0.2.2 with the parquet extension included.
+- It comes preinstalled with DuckDB ver 0.2.2 with the **parquet** extension included.
 - Has been tested to work with Linux and MacOS.
 - Currently supports NodeJS v12.17.0+.
 - Supports BIGINT and HUGEINT types as [BigInt](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt).
-- Provides an async API and a streaming one.
+- Provides a **Promise**-based API and a **Stream**-based one.
 
-## Using
+## Getting Started
 
-[See examples](https://github.com/deepcrawl/node-duckdb/tree/master/examples)
+### Installation
 
-## Developing
+To use Node-DuckDB in your project:
+
+```
+npm i node-duckdb
+```
+
+or
+
+```
+yarn add node-duckdb
+```
+
+Note: this will download the duckdb binary for your platform (currently Linux and MacOS are supported), or if it's not available will attempt to build it.
+
+### Usage
+
+Using node-duckdb is easy:
+
+```
+const db = new DuckDB();
+const connection = new Connection(db);
+await connection.execute("SELECT * FROM mytable;");
+```
+
+#### Promise API example
+
+An example using promises:
+
+```
+import { Connection, DuckDB } from "node-duckdb";
+
+async function queryDatabaseWithIterator() {
+  // create new database in memory
+  const db = new DuckDB();
+  // create a new connection to the database
+ const connection = new Connection(db);
+
+ // perform some queries
+ await connection.executeIterator("CREATE TABLE people(id INTEGER, name VARCHAR);");
+ await connection.executeIterator("INSERT INTO people VALUES (1, 'Mark'), (2, 'Hannes'), (3, 'Bob');");
+ const result = await connection.executeIterator("SELECT * FROM people;");
+
+ // fetch and print result
+ console.log(result.fetchAllRows());
 
-First build:
+ // release resources
+ connection.close();
+ db.close();
+}
 
-1. `yarn install` - installs dependencies including downloading duckdb
-2. `yarn build:ts` - builds typescript
-3. `yarn test` - runs all tests
+queryDatabaseWithIterator();
 
-Other useful scripts:
+```
 
-- `yarn build` - build everything
-- `yarn build:addon` - build just the bindings code
-- `yarn build:duckdb` - build just the duckdb database
-- `yarn build:ts` - build just the typescript code
-- `yarn lint` - lint the project
-- `yarn test` - run all tests
-- `yarn test csv` - run just the csv test suite
+#### Streaming API example
 
-## Publishing
+A demo of reading from DuckDB, transforming to CSV and writing to file using the streaming API:
+
+```
```

Comment:
Could we shorten these for the readme with link to full example? I imagine something like

```ts
const db = new DuckDB();
const connection = new Connection(db);

const resultStream = await connection.execute("SELECT * FROM people;");

resultStream.pipe(destinationStream);
```

Reply:
Also, do we actually need the semicolon in the statement?

Reply:
It works with and without it